### PR TITLE
Switch to .net8

### PR DESCRIPTION
**Pull request by marcin-dardzinski**
+ address some low hanging warnings

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "sdk": {
-    "version": "8.0.0-rc.2.23479.6",
-    "allowPrerelease": true,
+    "version": "8.0.0",
```
Comment (Saancreed) [inline global.json]: That's incorrect, should be `8.0.100`.

`8.0.0` is the runtime version.

Comment (marcin-dardzinski) [reply global.json]: Huh, was it incorrect all the time?

**Review by Saancreed — APPROVED** [review]

File: global.json (hunk 1 above)
Comment (Saancreed) [reply global.json]: Technically yes but 100 > 0 so it worked anyway :shrug:

**Review by Dragemil — APPROVED** [review]
Also there's MT 8.1.2 now.

**Comment by marcin-dardzinski** [comment]
> Also there's MT 8.1.2 now.

It introduced some breaking change + we still need to fix _non low hanging_ MT warnings, let's do it separately.

**Pull request merged**